### PR TITLE
Convert deployment components to TypeScript

### PR DESCRIPTION
**Pull request by spadgett**

### CODE REVIEW

File: frontend/public/components/deployment.tsx [hunk 1]
```diff
@@ -110,14 +115,41 @@ const environmentComponent = (props) => <EnvironmentPage
 />;
 
 const {details, editYaml, pods, envEditor, events} = navFactory;
-const DeploymentsDetailsPage = props => <DetailsPage
+export const DeploymentsDetailsPage: React.FC<DeploymentsDetailsPageProps> = props => <DetailsPage
   {...props}
+  kind="Deployment"
```
Comment (jhadvig) [inline frontend/public/components/deployment.tsx]: Let's declare `DeploymentsReference` and use that here and below ?
```ts
const DeploymentsReference: K8sResourceKindReference = 'Deployment';
```

**Comment by spadgett** [comment]
@jhadvig thanks, updated

**Comment by spadgett** [comment]
> ENOMEM: not enough memory, write

https://github.com/openshift/release/pull/3749

/test frontend

**Pull request merged**